js/main.js

```diff
@@ -1 +1,59 @@
+(function () {
+  function debounce(func, wait, immediate) {
+  	var timeout;
+  	return function() {
+  		var context = this, args = arguments;
+  		var later = function() {
+  			timeout = null;
+  			if (!immediate) func.apply(context, args);
+  		};
+  		var callNow = immediate && !timeout;
+  		clearTimeout(timeout);
+  		timeout = setTimeout(later, wait);
+  		if (callNow) func.apply(context, args);
+  	};
+  };
 
+  function getElementByClassName(className) {
+    if (document.getElementsByClassName) {
+      return document.getElementsByClassName(className)[0];
+    }
+
+    if (document.querySelectorAll) {
+      return document.querySelectorAll((' ' + className).replace(/ +/g, '.'))[0];
+    }
+  };
+
+  function addClass(elem, className) {
+    var re = new RegExp(className, 'g');
+    var elemClassName = elem.className;
+
+    if (re.test(elemClassName) === false) {
```

Comment:
It's not really booletproof. Try 'foo' on 'foobar' - it will be true and class won't be added